### PR TITLE
Bash functions cleaning

### PR DESCRIPTION
**Pull request by miniche**
This pull request will solve two issues:
- [x] #68 (Refactor BASH code in order to remove uneeded code)
- [x] #69 (Variables in the bash installer)

### CODE REVIEW

File: bash/functions.inc.sh [hunk 1]
```diff
@@ -39,13 +58,14 @@ function stop_cr(){
...
             logError "Error stopping CentralReport (Error code: ${RETURN_CODE})!"
-            return ${RETURN_CODE}
+            return 1
         fi
     fi
 
     return 0
 }
 
+# This function must be removed.
```
Comment (Ninir) [inline bash/functions.inc.sh]: Then remove it? :smile:

Comment (miniche) [inline bash/functions.inc.sh]: This is in the pull request https://github.com/CentralReport/CentralReport/pull/73 :P

File: bash/functions.inc.sh [hunk 2]
```diff
@@ -257,6 +328,14 @@ function delete_bin(){
     return 0
 }
 
+#
+# Removes the CentralReport library
+#
+# PARAMETERS: None
+# RETURN:
+#   0 = The CentralReport doesn't exist anymore
```
Comment (Ninir) [inline bash/functions.inc.sh]: To be consistent with other functions, I would write: `CentralReport library removed`

File: bash/functions.inc.sh [hunk 3]
```diff
@@ -366,6 +469,14 @@ function delete_log_directory(){
     return 0
 }
 
+#
+# Removes the PID directory
+#
+# PARAMETERS: None
+# RETURN:
+#   0 = The PID directory doesn't exist anymore
```
Comment (Ninir) [inline bash/functions.inc.sh]: PID directory removed

File: bash/functions.inc.sh [hunk 4]
```diff
@@ -588,6 +710,14 @@ function create_cr_user(){
     return 0
 }
 
+#
+# Deletes the CentralReport system user
+#
+# PARAMETERS: None
+# RETURN:
+#   0 = The user doesn't exist anymore
```
Comment (Ninir) [inline bash/functions.inc.sh]: User removed

File: bash/functions.inc.sh [hunk 5]
```diff
@@ -612,6 +742,13 @@ function remove_cr_user(){
     return 0
 }
 
+#
+# Deletes the CentralReport system group (only used for Mac OS X)
+#
+# PARAMETERS: None
+# RETURN:
+#   0 = The group doesn't exist anymore
```
Comment (Ninir) [inline bash/functions.inc.sh]: CentralReport system group removed

**Comment by Ninir** [comment]
:+1:

File: bash/functions.inc.sh (hunk 2 above)
Comment (miniche) [inline bash/functions.inc.sh]: I hesitated for these functions. In fact, the library can be already deleted, this function returns 0 if the library doesn't exist anymore (has been removed by this function or not existing on the system).

It's just a detail. If we prefer "CentralReport library removed", no problem.

**Comment by Guimove** [comment]
:+1:

**Comment by miniche** [comment]
Tested on Mac OS X 10.8, Debian 6 and Ubuntu 12.04. Will be merged in the next seconds.

**Pull request merged**